instrumentation/active_support/lib/opentelemetry/instrumentation/active_support/span_subscriber.rb

```diff
@@ -30,19 +30,21 @@ def self.subscribe(
 
         subscriber_object = ::ActiveSupport::Notifications.subscribe(pattern, subscriber)
 
-        ::ActiveSupport::Notifications.notifier.synchronize do
-          subscribers = ::ActiveSupport::Notifications.notifier.instance_variable_get(:@string_subscribers)[pattern]
-
-          if subscribers.nil?
-            OpenTelemetry.handle_error(
-              message: 'Unable to move OTEL ActiveSupport Notifications subscriber to the front of the notifications list which may cause incomplete traces.' \
-                       'Please report an issue here: ' \
-                       'https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/new?labels=bug&template=bug_report.md&title=ActiveSupport%20Notifications%20subscribers%20list%20is%20nil'
-            )
-          else
-            subscribers.unshift(
-              subscribers.delete(subscriber_object)
-            )
+        if Gem::Version.new(::Rails.version) < Gem::Version.new('7.2')
```

Comment:
Not sure what's best practice, but alternative would be to check if `synchronize` method is present to make it more like feature detection instead of version detection.

Reply:
I like that idea, I'll update it